extension.js

```diff
@@ -12,40 +12,66 @@ const activate = () => {
       return new Promise((resolve, reject) => {
         const luaFormatterScriptDir = __dirname + '/luacode';
         const luaFormatterScript = __dirname + '/luacode/formatter.lua';
-        const luaPath = vscode.workspace.getConfiguration('vscode-metalua-formatter').get('luaPath');
         const indentSize = vscode.workspace.getConfiguration('vscode-metalua-formatter').get('indentSize');
+        const platform = os.platform();
+        const arch = os.arch();
+        let builtInLuaPath;
 
-        if(!luaPath) {
-          vscode.window.showErrorMessage('Lua 5.1 path has not been specified in the configurations. Make sure to fill the \'luaPath\' field in the extensions settings.');
+        if(platform === "darwin") {
+          if(arch === "x64") {
+            builtInLuaPath = path.join(__dirname, 'lua', 'macos', 'lua51');
+          }
+          else if(arch === "arm64") {
+            builtInLuaPath = path.join(__dirname, 'lua', 'macos-arm64', 'lua51');
+          }
+        }
+        else if(platform === "linux") {
+          if(arch === "x64") {
+            builtInLuaPath = path.join(__dirname, 'lua', 'linux', 'lua51');
+          }
+          else if(arch === "arm64") {
+            builtInLuaPath = path.join(__dirname, 'lua', 'linux-arm64', 'lua51');
+          }
+        }
+        else if(platform === "win32") {
+          builtInLuaPath = path.join(__dirname, 'lua', 'windows', 'lua51.exe');
+        }
+        else {
+          vscode.window.showErrorMessage(`Platform '${platform}' is not supported.`);
           reject();
         }
-        else if(!fs.existsSync(luaPath)) {
+
+        const luaPath = vscode.workspace.getConfiguration('vscode-metalua-formatter').get('customLuaPath') || builtInLuaPath;
```

Comment:
What does `.get('customLuaPath')` return when the setting is left blank? Is it an empty string and that evaluates to `false`?

Reply:
It returns an empty string, which is falsy.